### PR TITLE
Add logging to serialization

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/storage/local_experiment.py [hunk 1]
```diff
@@ -200,4 +203,7 @@ def write_simulation_arguments(
         with open(
             self.mount_point / self._simulation_arguments_file, "w", encoding="utf-8"
         ) as f:
-            json.dump(dataclasses.asdict(info), f)
+            try:
+                json.dump(dataclasses.asdict(info), f)
+            except TypeError:
+                logger.error(f"Failed to serialize: {info}")
```
Comment (sondreso) [inline src/ert/storage/local_experiment.py:209]: This depends on this on `info `being serializable to text, but I guess this is an OK assumption?

**Review by sondreso — APPROVED** [review]

File: src/ert/storage/local_experiment.py (hunk 1 above)
Comment (sondreso) [reply src/ert/storage/local_experiment.py:209]: Also maybe not error?

Comment (sondreso) [reply src/ert/storage/local_experiment.py:209]: If we can continue I'll argue it's a warning

**Review by yngve-sk — APPROVED** [review]
LGTM

File: src/ert/storage/local_experiment.py (hunk 1 above)
Comment (oyvindeide) [reply src/ert/storage/local_experiment.py:209]: It was an error that caused us to notice it though, but we will work on a proper fix for this once we get some more data, with the goal of removing it altogether.

**Pull request merged**